plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -80,13 +81,22 @@ func (s *SchedulerServer) Run(_ []string) error {
 	}
 
 	go func() {
+		mux := http.NewServeMux()
 		if s.EnableProfiling {
-			http.HandleFunc("/debug/pprof/", pprof.Index)
-			http.HandleFunc("/debug/pprof/profile", pprof.Profile)
-			http.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+			mux.HandleFunc("/debug/pprof/", pprof.Index)
+			mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+			mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
 		}
-		http.Handle("/metrics", prometheus.Handler())
-		http.ListenAndServe(net.JoinHostPort(s.Address.String(), strconv.Itoa(s.Port)), nil)
+		mux.Handle("/metrics", prometheus.Handler())
+
+		server := &http.Server{
+			Addr:           net.JoinHostPort(s.Address.String(), strconv.Itoa(s.Port)),
+			Handler:        mux,
+			ReadTimeout:    5 * time.Minute,
```

Comment:
Those timeouts are probably unnecessary. However, the Addr & Handler were necessary (I mean I didn't know how to make both debug & metrics handlers). Will fix it.